pkg/process/scheduler/exclusive.go

```diff
@@ -0,0 +1,107 @@
+package scheduler
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/leptonai/gpud/pkg/log"
+	"github.com/leptonai/gpud/pkg/process"
+)
+
+var _ Scheduler = &exclusiveScheduler{}
+
+func NewExclusiveScheduler() Scheduler {
+	return &exclusiveScheduler{}
+}
+
+// exclusiveScheduler is a scheduler that runs a single process at a time.
+// Does not support concurrent execution of multiple processes.
+type exclusiveScheduler struct {
+	mu             sync.RWMutex
+	runningProcess process.Process
+}
+
+func (s *exclusiveScheduler) CountProcesses() (int, error) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	if s.runningProcess == nil {
+		return 0, nil
+	}
+	return 1, nil
+}
+
+// RunSyncMode starts a bash script, blocks until it finishes,
+// and returns the output and the exit code.
+// If there is already a process running, the previous process will be aborted,
+// and the new process will be started.
+func (s *exclusiveScheduler) RunSyncMode(ctx context.Context, script string) ([]byte, int32, error) {
+	s.stopRunningProcess(ctx)
```

Comment:
If we stop previsou porcess, it would not be a exclusive process. Shall we return an error if there is a process running?

Reply:
@cardyok ^ Our control plane needs to follow this semantics